### PR TITLE
[WIP] Fix bc stuck on network two

### PR DESCRIPTION
**Pull request by peterlimg**
# Fix: #85 
# Changes:
- Add worker for syncing missing nodes for latest finalized round when BC is stuck
- Make the MPT's iterator method cancelable.

### CODE REVIEW

File: code/go/0chain.net/core/util/merkle_patricia_trie.go [hunk 1]
```diff
@@ -873,6 +880,36 @@ func (mpt *MerklePatriciaTrie) UpdateVersion(ctx context.Context, version Sequen
...
+	handler := func(ctx context.Context, path Path, key Key, node Node) error {
+		if node == nil {
+			paths = append(paths, path)
+			keys = append(keys, key)
+		}
+		return nil
+	}
+
+	st := time.Now()
+	// TODO: may have dead lock for the iterate
```
Comment (kirillt) [inline code/go/0chain.net/core/util/merkle_patricia_trie.go:896]: Do we enter several times into iterating?

File: code/go/0chain.net/chaincore/chain/state_sync.go [hunk 2]
```diff
@@ -214,6 +245,35 @@ func (c *Chain) getStateNodes(ctx context.Context, keys []util.Key) (*state.Node
...
+func (c *Chain) getStateNodesFromSharders(ctx context.Context, keys []util.Key) (*state.Nodes, error) {
+	nsRequestor := StateNodesRequestor
+	params := &url.Values{}
+	for _, key := range keys {
+		params.Add("nodes", util.ToHex(key))
+	}
+	ctx, cancelf := context.WithCancel(common.GetRootContext())
+	var ns *state.Nodes
+	handler := func(ctx context.Context, entity datastore.Entity) (interface{}, error) {
+		rns, ok := entity.(*state.Nodes)
```
Comment (kirillt) [inline code/go/0chain.net/chaincore/chain/state_sync.go:257]: If "r" means "result" in `rns`, maybe just call variable "result"?

**Review by kirillt — COMMENTED** [review]
Only minor questions

File: code/go/0chain.net/chaincore/chain/entity.go [hunk 3]
```diff
@@ -1429,6 +1432,14 @@ func (c *Chain) callViewChange(ctx context.Context, lfb *block.Block) (
 	return c.viewChanger.ViewChange(ctx, lfb)
 }
 
+func (c *Chain) notifyToSyncFinalizedRoundState(bs *block.BlockSummary) {
+	select {
+	case c.syncLFBStateC <- bs:
+	case <-time.NewTimer(3 * time.Second).C:
```
[the remark this answers is not in the document]
Comment (peterlimg) [reply code/go/0chain.net/chaincore/chain/entity.go:1438]: No need to make it configurable as we don't need to change it.

File: code/go/0chain.net/chaincore/chain/state_sync.go (hunk 2 above)
Comment (kirillt) [reply code/go/0chain.net/chaincore/chain/state_sync.go:257]: We should encourage cleaner style.. All those abbreviations really spend time when you read code, especially for new people

**Pull request merged**